zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/FIFOScheduler.java

```diff
@@ -137,15 +138,26 @@ public void run() {
               listener.jobStarted(scheduler, runningJob);
             }
             runningJob.run();
+            Object jobResult = runningJob.getReturn();
             if (runningJob.isAborted()) {
               runningJob.setStatus(Status.ABORT);
+              LOGGER.debug("Job Aborted, " + runningJob.getId() + ", " +
+                  runningJob.getErrorMessage());
+            } else if (runningJob.getException() != null) {
+              LOGGER.debug("Job Error, " + runningJob.getId() + ", " +
+                  runningJob.getReturn());
+              runningJob.setStatus(Status.ERROR);
+            } else if (jobResult != null && jobResult instanceof InterpreterResult
+                && ((InterpreterResult) jobResult).code() == InterpreterResult.Code.ERROR) {
+              LOGGER.debug("Job Error, " + runningJob.getId() + ", " +
+                  runningJob.getReturn());
```

Comment:
is there more information about the ERROR that we could log here?
from jobResult?

Reply:
This if condition correpend to user code error, so the job return value could give sufficient info.